lua/roslyn/server.lua

```diff
@@ -47,16 +82,55 @@ function M.start_server(cmd, with_pipe_name)
             end
         end,
     }, function()
-        _pipe_name = nil
+        _pipe_names[config.root_dir] = nil
     end)
 end
 
-function M.stop_server()
-    if _server_object then
-        _server_object:kill(9)
+function M.stop_server(client_id)
+    local client = vim.lsp.get_client_by_id(client_id)
+
+    local server_object = _server_objects[client_id]
+    if server_object then
+        server_object:kill(9)
+    end
+
+    if client then
+        _pipe_names[client.root_dir] = nil
+    end
+end
+
+function M.start(config, opts)
+    opts = opts or {}
+    local reuse_client = opts.reuse_client or reuse_client_default
+    local bufnr = resolve_bufnr(opts.bufnr)
+
+    local all_clients = vim.lsp.get_clients()
+
+    for _, client in pairs(all_clients) do
+        if reuse_client(client, config) then
+            if vim.lsp.buf_attach_client(bufnr, client.id) then
+                _server_objects[client.id] = _current_server_object
```

Comment:
This is really identical to `vim.lsp.start`, but with some logic to save the `_current_server_object` inside `_server_objects` here so that we are able to stop it when deleting a buffer and such.

I think this is needed, but I should confirm it